### PR TITLE
Subnomo/project perms session

### LINKED ISSUE #27: Store project permissions in session [3]

### PR DESCRIPTION
**Pull request by subnomo**
Fixes #27 
## Changes in this PR.

- [X] Add user-accessible projects to session object
## Testing this PR.

1. Run `nosetests`
### Expected Output.

```
$ nosetests
.......................
----------------------------------------------------------------------
Ran 23 tests in 0.235s

OK
```

@osuosl/devs

### CODE REVIEW

**Comment by pop** [comment]
This needs to be documented. +1 pending that.

File: app/views.py [hunk 1]
```diff
@@ -42,6 +42,27 @@ def getUser():
...
+                           token=session['token'])
+    projects = ts.get_projects()
+
+    userProjects = []
+
+    # Get each project the user has access to
+    for project in projects:
+        if 'users' in project and username in project['users']:
+            userProjects.append(project)
+        elif 'owner' in project and username == project['owner']:
```
Comment (Kennric) [inline app/views.py]: I thought we deprecated the owner field on projects in favor of project level permissions (member, spectator, manager). I think we can drop this check.

File: app/views.py [hunk 2]
```diff
@@ -42,6 +42,27 @@ def getUser():
...
 
 
+def getProjects(username):
+    if not isLoggedIn():
+        return [{'error': "Not logged in."}]
+
+    ts = pymesync.TimeSync(baseurl=app.config['TIMESYNC_URL'],
+                           test=app.config['TESTING'],
+                           token=session['token'])
+    projects = ts.get_projects()
```
Comment (Kennric) [inline app/views.py:56]: I think this should only return projects that the current user has permissions on, in which case we can skip checking for the user in each project... @LyonesGamer can you confirm?

Comment (subnomo) [inline app/views.py:56]: @Kennric I had hoped so, but from my testing that doesn't seem to be true.

Comment (Kennric) [inline app/views.py:56]: Some day....

**Comment by Kennric** [comment]
This is looking pretty good to me. +1

**Comment by subnomo** [comment]
This branch is so far behind develop I think rebase would be too time-consuming, so I'm going to recreate this pull request.

**Pull request closed without merging**